programs/CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 set(programs_target "${MBEDTLS_TARGET_PREFIX}programs")
 add_custom_target(${programs_target})
 
+link_to_source(../framework/tests/programs/metatest.c)
```

Comment:
Sorry in https://github.com/Mbed-TLS/mbedtls-framework/issues/112, the part related to CMake is not correct. We use `link_to_source` for generated file that can be either in the source or in the build tree but here the moved files are just in the framework thus we just need to update their paths in the CMakeLists.txt files. Something like in: https://github.com/ronald-cron-arm/mbedtls/pull/new/move-programs-from-mbedtls-to-framework. Tested only on the default configuration.